### DISCUSSION TITLE
<CR> not behaving as expected in orgmode

### DISCUSSION THREAD

**Discussion by lyz-code** [LazyVim/LazyVim#2529] [Q&A] [answered]
With a clean install of LazyVim distribution when pressing `<CR>` from a heading it creates a new heading instead of moving the cursor to the body of the heading:

```org
* Test <--  press enter in insert mode
```
The result is:
```org
* Test
* <-- cursor here
```
The expected behaviour is:
```org
* Test
  <-- cursor here
```

As the [orgmode dev pointed out](https://github.com/nvim-orgmode/orgmode/issues/662) this happens even without nvim-orgmode installed.

My question then is how do I pinpoint which plugin is the culprit and how to solve it?

Thaanks!

**Comment by dpetka2001** [comment, marked as answer]
It's because of the `formatoptions`. If you do `:set fo-=r`, you will observe the difference. To make the change permanent, you should enforce it with an auto-command. I really have no idea what makes Neovim think that the character `*` is a comment leader in `.org` files.

**Reply by lyz-code** [reply]
I love you so much right now.... THANK YOU!

**Reply by lyz-code** [reply]
For those like me that don't know what `:set fo-=r` does [here is an explanation](https://vimhelp.org/change.txt.html#fo-table)

You can use the `formatoptions` option  to influence how Vim formats text.
'formatoptions` is a string that can contain any of the letters below.  The
default setting is "tcq".  The `r` option automatically inserts the current comment leader after pressing `<Enter>` in Insert mode.

The autocommand can be something like:

```lua
vim.api.nvim_create_autocmd("FileType", {
  pattern = "org",
  group = vim.api.nvim_create_augroup("orgmode", { clear = true }),
  callback = function()
    vim.opt.formatoptions:remove({ "r" })
  end,
})
```
